src/child/containers/sidebars/Sidebar.js

```diff
@@ -15,12 +15,49 @@ import favouritesShape from '../../propTypeShapes/favourites';
 class Sidebar extends Component {
     constructor(props) {
         super(props);
+        this.onDragOver = this.onDragOver.bind(this);
+        this.onDrop = this.onDrop.bind(this);
+        this.addActive = this.addActive.bind(this);
+        this.removeActive = this.removeActive.bind(this);
         this.focusFav = this.focusFav.bind(this);
         this.focusSearch = this.focusSearch.bind(this);
         this.toggleFavourite = this.toggleFavourite.bind(this);
         this.selectStock = this.selectStock.bind(this);
     }
 
+    onDragOver(e) {
+        if (!e.defaultPrevented && !this.props.sidebar.showFavourites) {
+            this.addActive();
+            e.preventDefault();
+        }
+    }
+
+    onDrop(e) {
+        const code = this.getCodeFromDT(e.dataTransfer.types);
+        const codes = this.props.favourites.codes;
+        if (!codes.includes(code)) {
+            this.props.dispatch(insertFavouriteAt(codes.length, code));
+        }
+        this.removeActive();
+    }
+
+    getCodeFromDT(types) {
+        for (let i = 0; i < types.length; i++) {
+            if (types[i] !== 'text/plain') {
+                return (types[i]).toUpperCase();
+            }
+        }
+        return undefined;
+    }
+
+    addActive() {
+        this.favourites.classList.add('active');
```

Comment:
I'm still not keen on manipulating the DOM directly like this (similar to [the comment in the hover styles PR](https://github.com/ScottLogic/StockFlux/pull/866#discussion_r71994300)) - I think using `props` or `state` would be more appropriate (for all instance of `classList.add` and `classList.remove` in this PR). Unless there's a reason to this approach?